### PR TITLE
Upgrade to Bootstrap v4 (fixes #46)

### LINKED ISSUE #46: Upgrade the CSS framework

### PR DESCRIPTION
**Pull request by ad0v0**
The task is done and waiting for your approval @aptiko

### CODE REVIEW

**Comment by ad0v0** [comment]
`040bca2` or `Merge branch 'master' into issue309-bootstrap4` commit resolves git conflicts

File: enhydris_synoptic/templates/enhydris-synoptic/groupstation-report.html [hunk 1]
```diff
@@ -3,23 +3,25 @@
...
+          {{ synts.title|default:synts.timeseries.name }}
+          {% if synts.subtitle %}
+            ({{ synts.subtitle }})
           {% endif %}
         </dt>
-        {% with precision=syntsg.timeseries_group.precision|default:0 %}
-          <dd>{{ syntsg.value|floatformat:precision }} {{ syntsg.timeseries_group.unit_of_measurement.symbol }}</dd>
+        {% with precision=synts.timeseries.precision|default:0 %}
+          <dd class="col-sm-10">{{ synts.value|floatformat:precision }} {{ synts.timeseries.unit_of_measurement.symbol
+            }}</dd>
```
Comment (aptiko) [inline enhydris_synoptic/templates/enhydris-synoptic/groupstation-report.html:24]: Apparently this is not allowed in Django. You can't break the line before the `}}`. The whole thing from `{{` to `}}` must be on the same line.

**Review by aptiko — CHANGES_REQUESTED** [review]
We generally don't merge; we rebase. So please rebase your two commits like this (meanwhile I've added two unrelated commits to master):

```
git fetch official
git checkout master
git merge --ff-only official/master
git checkout issue309-bootstrap4
git reset --hard 4aee235
git rebase master
```
(If you don't understand clearly what the `reset` does, it would be better to do this together with me.)

**Comment by aptiko** [comment]
Also please note that, unless the screen width is large, it doesn't show well.

Screenshot from 2020-09-21 16-24-00

Screenshot from 2020-09-21 16-24-40

Screenshot from 2020-09-21 16-25-08

**Comment by ad0v0** [comment]
@aptiko I've done rebasing before creating a pull request in order to make two commits (fix issue and fix code style). What wasn't done is fetching from official.

Now, fetching and fixing merge conflict are done as well, but I'm not able to push anything due to an error. Code:
```
λ git push --force-with-lease ad0v0 issue309-bootstrap4
Enter passphrase for key '/c/Users/Olga/.ssh/id_rsa':
To github.com:ad0v0/enhydris-synoptic.git
 ! [rejected]        issue309-bootstrap4 -> issue309-bootstrap4 (stale info)
error: failed to push some refs to 'git@github.com:ad0v0/enhydris-synoptic.git'
``` 
Screenshot: http://prntscr.com/unrhyc

Found similiar issue on stackoverflow (https://stackoverflow.com/questions/56191415/why-is-git-push-force-with-lease-failing-with-rejected-stale-info-even) where it says the issue can be caused by removed remote branch. Not sure about that because it still exists in my repo: https://github.com/ad0v0/enhydris-synoptic/tree/issue309-bootstrap4

Have you run into such issue before?

**Comment by aptiko** [comment]
Hi! Let's make a call with screen sharing for this. Today I'm available until about 16:00.

**Comment by ad0v0** [comment]
@aptiko I'm done with the feedback. 

By the way, is it fine, that after pulling the updated files, I've stopped seeing the full data with measurement (the way they look on your screenshots or on openmeteo)? Now I only see the Last update info: http://prntscr.com/uqnqyp

**Comment by aptiko** [comment]
The report wasn't being generated correctly because of some errors in the rebasing. I fixed those. You need to retrieve my changes, like this:
```
git remote add aptiko https://github.com/aptiko/enhydris-synoptic/  # This adds my repo as an additional remote
git fetch aptiko  # This fetches my repo
git checkout issue309-bootstrap4  # You are probably already there, but let's be certain
git reset --hard aptiko/issue309-bootstrap4
```
Remember that, in git, "branches" are really bad names. They aren't branches, they are labels, or pointers to commits. So the last command above moves the current branch (the label "issue309-bootstrap") to where the label "aptiko/issue309-bootstrap4" is also pointing to. If you look at the output of `git log --branches --remotes --oneline --decorate --graph` you'll understand it.

Now, I still have some observations:
1. There's a TODO somewhere. Please either do what you are supposed "to do" or remove the "TODO" comment. There must be no TODOs and no FIXMEs in the code.
2. The characters are now larger and the spacing in the report is also different. Is this the result of upgrading to Bootstrap v4? I.e. have the defaults of Bootstrap v4 changed? If yes, it's fine. But if not, please don't make changes. Let's keep this issue to upgrading to bootstrap v4.
3. There used to be a `max-width: 45em`, which you removed. Was that necessary because of the upgrade to Bootstrap v4? If not, please don't make this change. Let's keep this issue to upgrading to bootstrap v4.

**Comment by ad0v0** [comment]
I've put a comment `Make page centered and not too wide` that was initially in the existing code in a TODO message, as I didn't know if anyone needs it in the future. Okay, I'm going to remove it, as I don't need it as well

**Comment by ad0v0** [comment]
And a question: should I squash all commits into one so called `Implement feedback` when there's more than one changes request per pull request?

**Comment by aptiko** [comment]
Merged in 07e354d.

**Pull request closed without merging**